### ISSUE TITLE
remote sync controller

### ISSUE THREAD

**Issue by svub** [svub/lets-stay-in-touch#8]
TL;DR Checks all remote repositories, merges the info and updates the contact store.

Remote repositories are used to publish profile changes -- could be anything from Google Drive to IPFS. The updated info is encrypted individually for each of the user's contacts that this update is intended for by using their individual pub key. This way only the intended contact can decrypt the information and in turn update their local storage with the new information for this user.

This component will handle pushing and pull updates over all configured repositories via a plugin architecture. 

- [x] pushUpdate(): encrypt and publish profile to all contacts
- [x] pullUpdates(): get updates for all contacts from all repositories
- [x] enableRepository(id): configure a new repository

**Comment by ltfschoen** [comment]
what do you mean by "checking all the remote repositories"? is that just the database locations?

**Comment by svub** [comment]
yes exactly. I'll improve the description. A basic implementation is already done. Needs some testing though before closing the issue. And then we need some sort of background service that triggers the sync controller now and then.
